tracee/external/external.go

```diff
@@ -0,0 +1,95 @@
+package external
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+// Event is a user facing data structure representing a single event
+type Event struct {
+	Timestamp           float64    `json:"timestamp"`
+	ProcessID           int        `json:"processId"`
+	ThreadID            int        `json:"threadId"`
+	ParentProcessID     int        `json:"parentProcessId"`
+	HostProcessID       int        `json:"hostProcessId"`
+	HostThreadID        int        `json:"hostThreadId"`
+	HostParentProcessID int        `json:"hostParentProcessId"`
+	UserID              int        `json:"userId"`
+	MountNS             int        `json:"mountNamespace"`
+	PIDNS               int        `json:"pidNamespace"`
+	ProcessName         string     `json:"processName"`
+	HostName            string     `json:"hostName"`
+	EventID             int        `json:"eventId,string"`
+	EventName           string     `json:"eventName"`
+	ArgsNum             int        `json:"argsNum"`
+	ReturnValue         int        `json:"returnValue"`
+	Args                []Argument `json:"args"` //Arguments are ordered according their appearance in the original event
+}
+
+// Argument holds the information for one argument
+type Argument struct {
+	ArgMeta
+	Value interface{} `json:"value"`
+}
+
+// ArgMeta describes an argument
+type ArgMeta struct {
+	Name         string `json:"name"`
+	OriginalType string `json:"originalType"`
```

Comment:
Why OriginalType and not just Type?

Reply:
I originally called it just Type and then changed to OriginalType because when you look at the Arg struct, or the json of it, you see "name", "type" and "value" which implies "type" is the type of the value. but it's the type of the syscall arg. If you thing Type is better I will change it back, I also wasn't sure about this.

Reply:
Not sure I get your point here. Which Arg struct? You mean Argument struct?
This is ArgMeta struct, I don't think it will be confusing to call it Type here

Reply:
> You mean Argument struct?

yes

> This is ArgMeta struct

but ArgMeta is embedded within Argument. The idea is that we can use ArgMeta as is (for param tables) but also in Argument (for serialization). So if someone looks at the Argument struct, or the serialized Argument, they see "name" "type" "value". which implies the type describes the value in this struct, which is not true, it describes the original syscall value

Reply:
I don't see a problem with that. I think Type is fine